src/server/ClientHandler.java

```diff
@@ -20,29 +21,47 @@ public ClientHandler(Server server, Socket socket) {
         this.server = server;
         this.socket = socket;
         try {
+            socket.setSoTimeout(10000);
             inputStream = new DataInputStream(socket.getInputStream());
             outputStream = new DataOutputStream(socket.getOutputStream());
         } catch (IOException e) {
             e.printStackTrace();
         }
 
         //try to authenticate
-        while(true){
-            try{
+        while (true) {
+            try {
                 String str = inputStream.readUTF();
-                if(str.startsWith("/auth")){
+                if (str.startsWith("/auth")) {
                     String[] token = str.split("\\s");
                     nickName = server.auth(token[1], token[2]);
-                    if(nickName != null){
-                        System.out.printf("Client %s has been authorized", nickName);
+                    if (nickName != null) {
+                        System.out.printf("Client %s has been authorized \n", nickName);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Reply:
Thank you for your comment! Done :-)

Reply:
Happy to help.